### PR TITLE
feat(synthesizer): Support for MultiTable

### PR DESCRIPTION
**Pull request by portellaa**
```python
import os

os.environ['LOG_LEVEL'] = 'DEBUG'

os.environ['YDATA_BASE_URL'] = "fabric.dev.azure.ydata.ai"
os.environ["YDATA_TOKEN"] = "<TOKEN>"

from ydata.sdk.connectors import Connector
from ydata.sdk.datasources import DataSource
from ydata.sdk.synthesizers import MultiTableSynthesizer

datasource = DataSource.get('<datasource uid>')
print(datasource)

synthesizer = MultiTableSynthesizer(write_connector="<connector_uid>")
synthesizer.fit(datasource)

print("sample using connector from synthesizer class")
synthesizer.sample(1.5)

print("sample using a pre-fetched connector")
connector = Connector.get("e8e0443d-8cf4-47a6-a889-2ee29f19861e")
synthesizer.sample(2, write_connector=connector) # with optional write_connector
```

### CODE REVIEW

File: docs/examples/synthesizer_multitable.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Synthesize Multi Table
+
+**Use YData's *MultiTableSynthesizer* to generate multi table synthetic data from multiple RDBMS tables**
+
+Multi table is the way to synthesize data from multiple tables from a database, with a relational in mind...
+
```
Comment (portellaa) [inline docs/examples/synthesizer_multitable.md:6]: @fabclmnt can you help me with what to write here? 😅

File: docs/examples/synthesize_timeseries_data.md [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
+Timeseries is the most common type of data we encounter in data problems.
 
-When thinking about tabular data, we assume independence between different records, but this does not happen in reality. Suppose we check events from our day-to-day life, such as room temperature changes, bank account transactions, stock price fluctuations, and air quality measurements in our neighborhood. In that case, we might end up with datasets where measures and records evolve and are related through time. This type of data is known to be sequential or time-series data.
+When thinking about timeseries data, we assume independence between different records, but this does not happen in reality. Suppose we check events from our day-to-day life, such as room temperature changes, bank account transactions, stock price fluctuations, and air quality measurements in our neighborhood. In that case, we might end up with datasets where measures and records evolve and are related through time. This type of data is known to be sequential or time-series data.
 
 Thus, sequential or time-series data refers to any data containing elements ordered into sequences in a structured format.
```
[the remark this answers is not in the document]
Comment (aquemy) [reply docs/examples/synthesize_timeseries_data.md:9]: @portellaa this original text is correct.

**Review by aquemy — COMMENTED** [review]
The code is OK but I am concerned with the change of the typing from Optional to | None, or Union to | because this was introduced in Python 3.10.
If we want to support from Python 3.8 we cannot use it.

**Comment by portellaa** [comment]
@aquemy already reverted the changes to support python 3.8

**Review by portellaa — COMMENTED** [review]
reverted

**Review by aquemy — APPROVED** [review]

**Pull request merged**